tests/test_read_backed_experimental.py

```diff
@@ -0,0 +1,190 @@
+)
+
+
+@pytest.fixture(
+    params=[sparse.csr_matrix, sparse.csc_matrix, np.array, as_dense_dask_array],
+    ids=["scipy-csr", "scipy-csc", "np-array", "dask_array"],
+    scope="session",
+)
+def mtx_format(request):
+    return request.param
+
+
+@pytest.fixture(params=["zarr", "h5ad"], scope="session")
+def dskfmt(request):
+    return request.param
+
+
+@pytest.fixture(scope="session")
+def adata_remote_orig(
+    tmp_path_factory, dskfmt: str, mtx_format
+) -> tuple[AnnData, AnnData]:
+    orig_path = Path(tmp_path_factory.mktemp(f"orig.{dskfmt}"))
+    orig = gen_adata((1000, 1000), mtx_format)
+    orig.write_zarr(orig_path)
+    remote = read_backed(orig_path)
+    return remote, orig
+
+
+@pytest.fixture
+def adata_remote_with_store_tall_skinny(tmp_path_factory, mtx_format):
+    orig_path = Path(tmp_path_factory.mktemp("orig.zarr"))
+    M = 1000000  # forces zarr to chunk `obs` columns multiple ways - that way 1 access to `int64` below is actually only one access
+    N = 5
+    obs_names = pd.Index(f"cell{i}" for i in range(M))
+    var_names = pd.Index(f"gene{i}" for i in range(N))
+    obs = gen_typed_df(M, obs_names)
+    var = gen_typed_df(N, var_names)
+    orig = AnnData(
+        obs=obs,
+        var=var,
+        X=mtx_format(np.random.binomial(100, 0.005, (M, N)).astype(np.float32)),
+    )
+    orig.write_zarr(orig_path)
+    store = AccessTrackingStore(orig_path)
+    remote = read_backed(store)
+    return remote, store
+
+
+needs_xarray = pytest.mark.skipif(
+    not find_spec("xarray"), reason="Xarray is not installed"
+)
+
+
+@needs_xarray
+def test_access_count_obs_var(adata_remote_with_store_tall_skinny):
+    remote, store = adata_remote_with_store_tall_skinny
+    store.initialize_key_trackers(
+        ["obs/cat/codes", "obs/cat/categories", "obs/int64", "var/int64", "X"]
+    )
+    # a series of methods that should __not__ read in any data
+    remote.X  # the initial (non-subset) access to `X` should not read in data
+    remote.shape
+    remote.var
+    remote.obs
+    remote.obs["int64"]
+    remote.obs["int64"]
+    remote.obs["cat"]
+    assert store.get_access_count("obs/int64") == 0, store.get_subkeys_accessed(
+        "obs/int64"
+    )
+    assert (
+        store.get_access_count("obs/cat/categories") == 0
+    ), store.get_subkeys_accessed("obs/cat/categories")
+    subset = remote[
+        (remote.obs["cat"] == "a").data, :
+    ]  # `.data` for xarray, but should we handle internally?
+    subset.obs["int64"]
+    sub_subset = subset[0:10, :]
+    sub_subset.obs["int64"]
+    assert store.get_access_count("X") == 0, store.get_subkeys_accessed("X")
+    assert store.get_access_count("obs/int64") == 0, store.get_subkeys_accessed(
+        "obs/int64"
+    )
+    assert store.get_access_count("var/int64") == 0, store.get_subkeys_accessed(
+        "var/int64"
+    )
+    # all codes read in for subset
+    assert store.get_access_count("obs/cat/codes") == 4, store.get_subkeys_accessed(
```

Comment:
that means “4” because there’s 4 codes?

Reply:
4 because there are 4 chunks under `obs/cat/codes`

Reply:
```suggestion
    # all codes read in for subset (from 4 chunks)
    assert store.get_access_count("obs/cat/codes") == 4, store.get_subkeys_accessed(
```